### PR TITLE
remove polarity from memory domain

### LINKED ISSUE #276: Remove polarity from memory domains
With the forward (strongest postcondition) verifier, the polarity of all memory domains is always negative (i.e. memory domains only accumulate values that are not equivalent). By committing to only supporting negative-polarity domains, we can significantly simplify their implementation.

### PR DESCRIPTION
**Pull request by danmatichuk**
fixes #276 and #204

### CODE REVIEW

File: src/Pate/Equivalence/MemoryDomain.hs [hunk 1]
```diff
@@ -49,26 +49,23 @@ import qualified Pate.Parallel as Par
...
+-- all addresses are considered to be "in" the domain unless they are
+-- present in the given 'PMC.MemCellPred'.
+-- Notably this does not consider the memory region covered by a
+-- given 'PMC.MemCell'. A cell must be exactly included in the
+-- underlying 'PMC.MemCellPred' to be excluded by the domain -
+-- it is not sufficient for a cell to be subsumed by an entry.
 --
--- The interpretation of those predicates is subject to the 'memDomainPolarity'.
+-- This is sound, less precise, and significantly easier to
+-- prove membership.
```
Comment (travitch) [inline src/Pate/Equivalence/MemoryDomain.hs:61]: What is it less precise than, or significantly easier to prove membership compared to?

**Review by travitch — APPROVED** [review]
Love it. Just had one clarification request in there

File: src/Pate/Equivalence/MemoryDomain.hs [hunk 2]
```diff
@@ -156,68 +159,44 @@ containsCell ::
...
+universal _sym = MemoryDomain (PMC.MemCellPred M.empty)
 
 -- | Derive a 'MemoryDomain' from a set of 'MT.MemFootprint'.
--- The semantics of this domain depend on the given polarity: for positive polarity
--- the resulting domain includes all of the given footprints. For negative polarity,
--- the resulting domain includes all of memory, except for the given footprints.
--- TODO: This also filters the set of footprints according to the given polarity.
--- For positive polarity, the footprints are filtered to only contain the reads,
--- while negative polarity filters only the writes. This should be factored into
--- a separate step (see https://github.com/GaloisInc/pate/issues/204)
```
Comment (travitch) [inline src/Pate/Equivalence/MemoryDomain.hs:177]: Does this close #204 also?

File: src/Pate/Equivalence/MemoryDomain.hs (hunk 1 above)
Comment (danmatichuk) [reply src/Pate/Equivalence/MemoryDomain.hs:61]: Good question. I moved this discussion to the function where it is relevant and expanded on it a bit.

**Pull request merged**